health/polling.go

```diff
@@ -157,35 +157,38 @@ func (r *pollingChecker) New(
 		defer ticker.Stop()
 
 		for {
-			ctx, cancel := context.WithTimeout(ctx, r.timeout)
-			defer cancel()
-
-			result := r.prober.Probe(ctx, conn)
-
-			lastState := state
-			switch {
-			case result == StateHealthy && (state == StateUnhealthy || state == StateDegraded || state == StateUnknown):
-				counter++
-				if counter >= r.healthyThreshold {
+			func() {
+				ctx, cancel := context.WithTimeout(ctx, r.timeout)
+				defer cancel()
```

Comment:
Previously this was within a `for` loop which I thought usually lint would catch, but by creating a new `func` this should be scoped properly now